analyses/fusion_filtering/02-fusion-filtering.R

```diff
@@ -0,0 +1,241 @@
+  # @param spanningFragCountFilter An integer threshold for (SpanningFragCount - JunctionReadCount)
+  # @param expressionFilter An integer threshold for TPM/FPKM filter
+  # @return Standardized fusion calls filtered to pass QC and remove calls with insufficient read-support and annotation red-flags
+  
+  if( reathroughFilter!=FALSE & nrow(standardFusioncalls[grep('read.*through|NEIGHBORS',standardFusioncalls$annots,ignore.case = TRUE),]) > 0){
+    # Gather read throughs from standardized fusion calls  
+    rts <- standardFusioncalls[grep('read.*through|NEIGHBORS',standardFusioncalls$annots,ignore.case = TRUE),"FusionName"]
+    # Reverse of read throughs to capture 
+    rts.rev <- unique(unlist(lapply(strsplit(rts, '--'), FUN = function(x) paste0(x[2],'--',x[1]))))
+    # Combine read through and reverse fusion genes 
+    rts <- unique(c(rts, rts.rev))
+    standardFusioncalls <- standardFusioncalls[-which(standardFusioncalls$FusionName %in% rts),]
+  }
+
+  if( !missing(readingFrameFilter ) ){
+    standardFusioncalls <- standardFusioncalls[grep(readingFrameFilter,standardFusioncalls$Fusion_Type),]
+  }
+  
+  if( !missing(artifactFilter) & nrow(standardFusioncalls[grep(artifactFilter,standardFusioncalls$annots),])>0 ) {
+    standardFusioncalls <- standardFusioncalls[-grep(artifactFilter,standardFusioncalls$annots),]
+  }
+
+  if( !missing(junctionReadCountFilter ) ){
+    standardFusioncalls <- standardFusioncalls[which(standardFusioncalls$JunctionReadCount>junctionReadCountFilter),]
+  }
+
+  if( !missing(spanningFragCountFilter ) ){
+    # false positive calls either at the start or end of a transcript will be supported by an uneven majority of support from spanning fragments compared to junction reads
+    # to remove these calls we are implementing this condition below
+    standardFusioncalls <- standardFusioncalls[which( (standardFusioncalls$SpanningFragCount- standardFusioncalls$JunctionReadCount ) < spanningFragCountFilter),]
+  }
+  
+  return(standardFusioncalls)
+}
+
+# QC filter: artifact and read support
+QCFiltered<-fusion_filtering_QC(standardFusioncalls = standardFusioncalls,junctionReadCountFilter = junctionReadCountFilter,spanningFragCountFilter = spanningFragCountFilter,readingFrameFilter = readingFrameFilter,artifactFilter = artifactFilter,reathroughFilter = reathroughFilter)
+
+
+# load expressionMatrix RDS for expression based filtering for less than given threshold
+expressionMatrix<-readRDS(expressionMatrix)
+expressionMatrix <- cbind(expressionMatrix, colsplit(expressionMatrix$gene_id, pattern = '_', names = c("EnsembleID","GeneSymbol")))
+
+
+fusion_filtering_expression<-function(Sample=Sample,standardFusioncalls=standardFusioncalls,expressionFilter=expressionFilter){
+  print(Sample)  
+  #gene region fusions
+  standardFusioncallsGeneRegion<-standardFusioncalls[which(standardFusioncalls$Gene2A=="" & standardFusioncalls$Gene2B=="") ,]  
```

Comment:
So every gene column [1A,2A,2B,1B] will have some value depending on genic( Gene1A and Gene1B) and Gene2[A|B] will be additionally populated depending on the intergenic calls. We want to check and keep all fusions where the genes are expressed more than the expressionFilter threshold.

Reply:
If I had a row that had Gene1A, Gene1B, Gene2A, and Gene2B all populated (don't know if both can be intergenic) - I would want the expression values for all 4 genes to be greater than the filter threshold, correct?

Reply:
Correct, we would want all expression values to be greater than threshold. Since all the expression values will be saved in the column with the sample id I'm looking for this condition which(as.integer(standardFusioncalls[,Sample]) > expressionFilter)